bucket/leaky_bucket.go

```diff
@@ -40,61 +49,37 @@ func (b *LeakyBucket) StopDripping() {
 	}()
 }
 
-func (b *LeakyBucket) UnsafeAvailable() int {
-	return b.available
+func (b *LeakyBucket) Available() int {
+	return len(b.bucket)
```

Comment:
Can you tell I don't always read all the docs for a language I'm learning?
:P

On Fri, Dec 18, 2015 at 10:41 AM, Jakob Külzer <notifications@github.com>
wrote:

> In bucket/leaky_bucket.go
> <https://github.com/Shopify/themekit/pull/127#discussion_r48037085>:
>
> > @@ -40,61 +49,37 @@ func (b *LeakyBucket) StopDripping() {
> >  	}()
> >  }
> >
> > -func (b *LeakyBucket) UnsafeAvailable() int {
> > -	return b.available
> > +func (b *LeakyBucket) Available() int {
> > +	return len(b.bucket)
>
> https://golang.org/pkg/builtin/#len :
>
> The len built-in function returns the length of v, according to its type:
> Channel: the number of elements queued (unread) in the channel buffer;
>
> :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/themekit/pull/127/files#r48037085>.
>



-- 
Chris Saunders // Developer @ Shopify